routes/projects/files/rename.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * PUT /projects/:username/:projectName/files/:pathToFile/rename
+ * The :pathToFile parameter must be URI encoded
+ * Renames the specified file with the new name given in the message body
+ * {
+ *   name: <string> required
+ * }
+*/
+
+var renameSchema = require('../../../schemas.js').rename;
+
+module.exports = function(server) {
+  server.route({
+    method: 'PUT',
+    path: '/projects/{username}/{projectName}/files/{path}/rename',
```

Comment:
this is an interesting question, but i think we need to remember that a path and the name of a file are actually the same, e.g:

`/public/styles/custom.css` is the true name of `custom.css`